### PR TITLE
Feature/role based access system

### PR DESCRIPTION
**Pull request by vc-jinal**
changed URLs of routes

### CODE REVIEW

File: role-based-access-system/src/app.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import express from "express";
+import permissionRoutes from "./routes/permission.route.js";
+import userRoutes from "./routes/user.route.js";
+import rolesRoutes from "./routes/roles.route.js";
+
+const app = express();
+app.use(express.json());
+
+app.use("/api/permissions", permissionRoutes);
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/app.js:9]: Where is the index file ?

File: role-based-access-system/src/routes/roles.route.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            return res.send({ statusCode: 404, message: "Data not found" });
+        }
+        return res.send({ statusCode: 200, message: "Permissions fetched successfully", data: permission });
+    } catch (error) {
+        return res.send({ statusCode: 500, message: "Internal Server Error" });
+    }
+});
+
+//add permission to role
+route.post('/:roleId/permission/:permissionId', async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/roles.route.js:20]: In post request values should be part of body

**Review by gaurangviitorcloud — CHANGES_REQUESTED** [review]

File: role-based-access-system/src/routes/roles.route.js (hunk 2 above)
Comment (vc-jinal) [reply role-based-access-system/src/routes/roles.route.js:20]: changed request values in body

File: role-based-access-system/src/app.js (hunk 1 above)
Comment (vc-jinal) [reply role-based-access-system/src/app.js]: created index.js

**Review by vc-mohit — APPROVED** [review]

File: role-based-access-system/src/routes/roles.route.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { Router } from "express";
+const route = Router();
+import query from "../db/query.js";
+
+// get all permissions for a role.
+route.get("/:roleId/permission", async (req, res) => {
+    try {
+        const roleId = req.params.roleId;
+        const permission = await query(`SELECT p.name FROM permission p INNER JOIN role_permission rp ON rp.permission_id = p.id WHERE rp.role_id =${roleId}`);
+        if (permission.affectedRows === 0) {
```
[the remark this answers is not in the document]
Comment (vc-jinal) [reply role-based-access-system/src/routes/roles.route.js]: changed it

**Review by gaurangviitorcloud — APPROVED** [review]

**Pull request merged**